phylokit/distance.py

```diff
@@ -124,18 +124,19 @@ def kc_distance(ds1, ds2, lambda_=0.0):
     return np.linalg.norm((1 - lambda_) * (m[0] - m[1]) + lambda_ * (M[0] - M[1]))
 
 
-def _get_node_partition(postorder, left_child, right_sib):
-    tree_sets = [set()] * (len(left_child) - 1)
+@jit.numba_njit
```

Comment:
I think this breaks for the general case where the leaves are not 0,..., n - 1 though.

Can we get a few tests in for this case? Plus, I think it would be good to keep the simple set based version in the test module, as a "definition" case.

Reply:
I see, I'll make a few test cases on this one.

Reply:
Having a test case now, but for now, I am just adding nodes and edges manually, not sure if there's an effective way of doing this.